src/gateway/routers/Aggregator.sol

```diff
@@ -218,4 +278,14 @@ contract Aggregator is Auth, IAggregator {
     function votes(bytes32 messageHash) external view returns (uint16[8] memory) {
         return messages[messageHash].votes;
     }
+
+    /// @inheritdoc IAggregator
+    function estimate(bytes calldata payload) external view returns (uint256 estimated) {
```

Comment:
Wdyt about moving this to `Gateway` instead? It seems to fit more intuitively there. Would be a bit more complex to get the routers but should be possible.

Reply:
Main reason why I left everything in the aggregator is because routers are there.
Getting the routers won't be an issue since they are a public variable.
What bothers me is that in order to make this work I have to also provide `GasService` here as well.
You basically get duplication of "functionality". Why not just keep it in the aggregator?

On second thought... Since this will be a view function, what stops us to create smth like this.

```solidity
contract Gateway {

 //......
 
   function estimate() external view returns (uint256) {
       return AggregatorLike(aggregator).estimate();
    }
 }
 ```

Reply:
Maybe we should actually keep the `estimate` implementation in aggregator, and add a view method like the above to the `CentrifugeRouter`. So that is a "one stop shop" for interacting with Centrifuge

Reply:
This second suggestion is actually better. Indeed we would like to make this abstracted as much as possible and make the integrator use only `CentrifugeRouter`.